public/js/eventsPage.js

```diff
@@ -208,24 +210,55 @@ async function displayHiringPartnersAttending(event){
  *
  * @param events an object which contains information about an event
  */
-async function eventGenerator(event, hiringPartners) {
+async function eventGenerator(event, hiringPartners, applicants) {
     let eventInformation = ''
     let date = new Date(event.date).toDateString()
+    let numberOfAttendees
+    if (event.category_name === 'Assessment') {
+        numberOfAttendees = applicants.length
+        console.log(numberOfAttendees)
+    }
+    let headerDate = new Date(event.date).toDateString().substring(0,3) + ' ' + new Date(event.date).getUTCDate() + '/' + new Date(event.date).getUTCMonth() + '/' + new Date(event.date).getUTCFullYear()
     eventInformation +=
         `<div class="event">
-        <div class="header">
-            <h4>${event.name} - ${date}</h4>
-            <button class="show-event-info btn btn-primary" data-reference='${event.id}'>More Info</button>
+        <div class="header header-show-event-info" data-reference='${event.id}'>
+            <h4>${event.name} - <span class="headerDateTime">${headerDate} Time: ${event.start_time.slice(0, -3)} - ${event.end_time.slice(0, -3)} </span><span class="badge">${event.category_name}</span>`
+    if (event.category_name === 'Assessment') {
+        eventInformation += `<span class="badge">Attendees: ${numberOfAttendees}</span>`
+    }
+    eventInformation +=
+            `</h4> 
+            <button class="show-event-info btn" data-button-id='button${event.id}' data-reference='${event.id}'>More Info</button>
         </div>
         <div id="moreInfo${event.id}" class="hidden moreInfo">
-        <p>Event Category: ${event.category_name}</p>
-        <p>Date: ${date}</p>
-        <p>Location: ${event.location}</p>
-        <p>Start Time: ${event.start_time.slice(0, -3)}</p>
-        <p>End Time: ${event.end_time.slice(0, -3)}</p>`
+        <p><span>Event Category:</span> ${event.category_name}</p>
+        <p><span>Date:</span> ${date}</p>
+        <p><span>Location:</span> ${event.location}</p>
+        <p><span>Start Time:</span> ${event.start_time.slice(0, -3)}</p>
+        <p><span>End Time:</span> ${event.end_time.slice(0, -3)}</p>`
 
     if (event.notes !== null) {
-        eventInformation += `<p>Notes: ${event.notes}</p>`
+        eventInformation += `<p><span>Notes:</span> ${event.notes}</p>`
+    }
+
+    if (event.category_name === 'Assessment') {
+        // put code for assessment event type here!
+        eventInformation += '<div>';
+        eventInformation += '<table class="col-xs-12 table-bordered table">';
+        eventInformation += '<tr>';
+        eventInformation += '<th class="col-xs-2">Name</th>';
+        eventInformation += '<th class="col-xs-3">Email</th>';
+        eventInformation += '</tr>';
+            applicants.forEach((applicant) => {
+                if (applicant.assessmentDay == event.id) {
```

Comment:
I think you need a check like this when counting applicants, but again this probs falls under s10?

Reply:
Discussed but have added this in the new commit as the S10 query to the DB doesn't currently only fetch applicants attending a given event